src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,30 +1,74 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import React from 'react';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+interface Props {
+  query: string;
+  status: 'all' | 'active' | 'completed';
+  onQueryChange: (value: string) => void;
+  onStatusChange: (value: 'all' | 'active' | 'completed') => void;
+}
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
-      </span>
-    </p>
-  </form>
-);
+export const TodoFilter: React.FC<Props> = ({
+  query,
+  status,
+  onQueryChange,
+  onStatusChange,
+}) => {
+  const handleStatusChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const newStatus = event.target.value;
+
+    if (
+      newStatus === 'all' ||
+      newStatus === 'active' ||
+      newStatus === 'completed'
+    ) {
+      onStatusChange(newStatus);
+    } else {
+      console.error(`Invalid status value: ${newStatus}`); // eslint-disable-line no-console
```

Comment:
While logging an error to the console is useful for debugging, consider providing user feedback if an invalid status value is detected. This could be a message displayed on the UI.